File: src/backend/core/internal/InternalTypes.cs
namespace SnippetVault {
using System;

public class AuthenticateIntern {
    public int userId {get; set;}
    public string salt {get; set;}
    public string hash {get; set;}
    public DateTime expiration {get; set;}
    public string accessToken {get; set;}
}

public class AuthorizeIntern {
    public string accessToken {get; set;}
    public DateTime expiration {get; set;}
}

public class SnippetIntern {
    public int taskLanguageId {get; set;}
    public string content {get; set;}
    public int status {get; set;}
    public int score {get; set;}
    public string libraries {get; set;}
}

public class UserNewIntern {
    public string userName;
    public string hash;
    public string salt;
    public string accessToken;
    public DateOnly dtExpiration;
}

public class TaskCUIntern {
    public int taskGroupId {get; set;}
    public string taskGroupName {get; set;}
    public string name {get; set;}
    public string description {get; set;}
    public int existingId {get; set;}
    public bool isDeleted {get; set;}
}

public enum SnippetStatus {
    Proposal = 1,
    Declined = 2,
    Approved = 3,
}

}

File: src/backend/main/tech/sozonov/SnippetVault/cmn/core/internal/InternalTypes.java
package tech.sozonov.SnippetVault.cmn.core.internal;
import java.time.LocalDate;
import java.time.LocalDateTime;

import lombok.Builder;


public class InternalTypes {

public static class AuthenticateIntern {
    public int userId;
    public String salt;
    public String hash;
    public LocalDateTime expiration;
    public String accessToken;
}

public static class AuthorizeIntern {
    public String accessToken;
    public LocalDateTime expiration;
}

public static class SnippetIntern {
    public int taskLanguageId;
    public String content;
    public int status;
    public int score;
    public String libraries;
}

@Builder
public static class UserNewIntern {
    public String userName;
    public String hash;
    public String salt;
    public String accessToken;
    public LocalDate dtExpiration;
}

public static class TaskCUIntern {
    public int taskGroupId;
    public String taskGroupName;
    public String name;
    public String description;
    public int existingId;
    public boolean isDeleted;
}

public static class  SnippetStatus {
    public static final int Proposal = 1;
    public static final int Declined = 2;
    public static final int Approved = 3;
}

}